tests/test_ai_inserts.py

```
@@ -0,0 +1,363 @@
+        result = ai_service._validate_title_guardrails(new_title, original_title)
+        
+        assert result is True
+    
+    def test_validate_title_guardrails_missing_daily_devlog(self, ai_service):
+        """Test title guardrail validation with missing 'Daily Devlog'."""
+        new_title = "Just Some Title — Jan 15, 2025"
+        original_title = "Daily Devlog — Jan 15, 2025"
+        
+        result = ai_service._validate_title_guardrails(new_title, original_title)
+        
+        assert result is False
```

Ruff (0.12.2)

Use of `assert` detected

(S101)